redis/src/sentinel.rs

```diff
@@ -807,3 +837,251 @@ impl SentinelClient {
+                    ref mut insecure,
+                    ref mut tls_params,
+                } => {
+                    if let Some(tls_mode) = self.client_to_sentinel_params.tls_mode {
+                        match tls_mode {
+                            TlsMode::Secure => {
+                                *insecure = false;
+                            }
+                            TlsMode::Insecure => {
+                                *insecure = true;
+                            }
+                        }
+                    }
+                    if let Some(certs) = self.client_to_sentinel_params.certificates.clone() {
+                        let new_tls_params = retrieve_tls_certificates(certs)?;
+                        *tls_params = Some(new_tls_params);
+                    }
+                }
+                ConnectionAddr::Unix(_) => {
+                    if self.client_to_sentinel_params.tls_mode.is_some() {
+                        return Err(RedisError::from((
+                            ErrorKind::InvalidClientConfig,
+                            "Tls mode cannot be set for unix sockets.",
+                        )));
+                    }
+                    if self.client_to_sentinel_params.certificates.is_some() {
+                        return Err(RedisError::from((
+                            ErrorKind::InvalidClientConfig,
+                            "Certificates cannot be set for unix sockets.",
+                        )));
+                    }
+                }
+            }
+        }
+
+        let mut client_to_sentinel_redis_connection_info = RedisConnectionInfo::default();
+
+        if let Some(db) = self.client_to_sentinel_params.db {
+            client_to_sentinel_redis_connection_info.db = db;
+        }
+
+        if let Some(username) = self.client_to_sentinel_params.username.as_ref() {
+            client_to_sentinel_redis_connection_info.username = Some(username.clone());
+        }
+
+        if let Some(password) = self.client_to_sentinel_params.password.as_ref() {
+            client_to_sentinel_redis_connection_info.password = Some(password.clone());
+        }
+
+        if let Some(protocol) = self.client_to_sentinel_params.protocol {
+            client_to_sentinel_redis_connection_info.protocol = protocol;
+        }
+
+        let sentinels = self
+            .sentinels
+            .into_iter()
+            .map(|connection_addr| ConnectionInfo {
+                addr: connection_addr,
+                redis: client_to_sentinel_redis_connection_info.clone(),
+            })
+            .collect();
+
+        SentinelClient::build(
+            sentinels,
+            self.service_name,
+            Some(sentinel_to_redis_connection_info),
+            self.server_type,
+        )
+    }
+
+    /// Set tls mode for the connection between sentinels and redis nodes
+    pub fn sentinel_to_redis_tls_mode(mut self, tls_mode: TlsMode) -> SentinelClientBuilder {
```

Comment:
I know that the crate is inconsistent on this, I'm trying to make it better. please use `set_` on all of these setters.